extensions/amp-gfycat/0.1/amp-gfycat.js

```diff
@@ -48,35 +75,205 @@ class AmpGfycat extends AMP.BaseElement {
   }
 
   /** @override */
-  layoutCallback() {
-    const gfyid = user().assert(
+  buildCallback() {
+    this.videoid_ = this.getVideoId_();
+
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+
+    installVideoManagerForDoc(this.element);
```

Comment:
I assume the point of this was for AMP to manage your autoplay, in that case, you need to always tell your player to turn off autoplay and here you need to add `autoplay` attribute if `noautoplay` is not specified. e.g.
```
if(!this.element.hasAttribute('noautoplay')) {
  this.element.setAttribute('autoplay', '');
}
```

this must happen before registering with video-manager, then video-manager will take over autoplay behaviour for you by calling `play` `pause` commands at the right times